File: doordeck-sdk/src/mingwMain/resources/csharp/Model/Responses/HelperResponses.cs
namespace Doordeck.Headless.Sdk.Model.Responses;

public class AssistedLoginResponse
{
    public required bool RequiresVerification { get; set; }
    public required bool RequiresRetry { get; set; }
}

public class AssistedRegisterEphemeralKeyResponse
{
    public required bool RequiresVerification { get; set; }
    public required bool RequiresRetry { get; set; }
}

File: doordeck-sdk/src/androidMain/kotlin/com/doordeck/multiplatform/sdk/model/responses/HelperResponses.kt
package com.doordeck.multiplatform.sdk.model.responses

data class AssistedLoginResponse(
    val requiresVerification: Boolean,
    val requiresRetry: Boolean
)

data class AssistedRegisterEphemeralKeyResponse(
    val requiresVerification: Boolean,
    val requiresRetry: Boolean
)

internal fun NetworkAssistedLoginResponse.toAssistedLoginResponse(): AssistedLoginResponse = AssistedLoginResponse(
    requiresVerification = requiresVerification,
    requiresRetry = requiresRetry
)

internal fun NetworkAssistedRegisterEphemeralKeyResponse.toAssistedRegisterEphemeralKeyResponse(): AssistedRegisterEphemeralKeyResponse = AssistedRegisterEphemeralKeyResponse(
    requiresVerification = requiresVerification,
    requiresRetry = requiresRetry
)